packages/server/modules/core/helpers/scanTable.ts

```diff
@@ -0,0 +1,17 @@
+import { Knex } from 'knex'
+
+export const scanTableFactory = <TRecord extends object>({
+  db
+}: {
+  db: Knex<TRecord>
+}) =>
+  async function* ({ tableName, batchSize }: { tableName: string; batchSize: number }) {
+    const [rowsCount] = await db(tableName).count()
+    const MAX_LIMIT = parseInt(rowsCount.count.toString())
+    let offset = 0
+
+    while (offset <= MAX_LIMIT) {
+      yield db<TRecord>(tableName).limit(batchSize).offset(offset)
```

Comment:
This could miss out on some rows if there are any created after `.count()` but before `.select()`. Instead of looping until MAX_LIMIT, loop until the db query no longer returns any rows. And then on top of that, add a failsafe that's practically uncreachable (I dunno, batchSize * 1000) to ensure it stops somewhere even in broken/infinite loop scenarios.